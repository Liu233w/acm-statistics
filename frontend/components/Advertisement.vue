<template>
  <div :id="id" />
</template>

<script>
export default {
  props: {
    id: {
      type: String,
      required: true,
    },
    size: {
      type: String,
      required: false,
      default: '728x90',
    },
  },
  mounted() {
    try {
      window._mNHandle.queue.push(function () {
        window._mNDetails.loadTag(this.id, this.size, this.id)
      })
    } catch (error) {
      // ignore
    }
  },
}
</script>
