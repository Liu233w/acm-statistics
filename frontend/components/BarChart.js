import { Bar, mixins } from 'vue-chartjs'

const { reactiveProp } = mixins

export default {
  extends: Bar,
  mixins: [reactiveProp],
  data() {
    return {
      options: {
        responsive: true,
      },
    }
  },
  mounted() {
    this.renderChart(this.chartData, this.options)
  },
}
