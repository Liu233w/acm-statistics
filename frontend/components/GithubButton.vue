<template>
  <v-tooltip
    left
    v-resize="onResize"
  >
    <template #activator="{ on }">
      <v-btn
        icon
        v-on="on"
        href="https://github.com/Liu233w/acm-statistics"
        target="_blank"
        rel="noopener noreferrer"
        v-show="show"
      >
        <v-icon medium>
          fab fa-github
        </v-icon>
      </v-btn>
    </template>
    <span>Star me on Github!</span>
  </v-tooltip>
</template>

<script>
export default {
  data() {
    return {
      show: true,
    }
  },
  mounted() {
    this.onResize()
  },
  methods: {
    onResize() {
      const width = window.innerWidth
      if (width < 450) {
        this.show = false
      } else {
        this.show = true
      }
    },
  },
}
</script>
