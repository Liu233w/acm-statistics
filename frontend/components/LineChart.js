import { Line, mixins } from 'vue-chartjs'

const { reactiveProp } = mixins

export default {
  extends: Line,
  mixins: [reactiveProp],
  props: {
    options: {
      type: Object,
      default: {
        responsive: true,
        maintainAspectRatio: false,
      },
    },
  },
  mounted() {
    this.renderChart(this.chartData, this.options)
  },
}
