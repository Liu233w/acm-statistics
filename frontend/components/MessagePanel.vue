<template>
  <v-expand-transition>
    <v-card
      v-show="show"
      class="message-panel"
    >
      <v-slide-y-transition
        group
        tag="v-list"
        class="pa-0"
      >
        <v-list-item
          v-for="(item,i) in list"
          :key="item.id"
          class="pa-0"
        >
          <v-alert
            dismissible
            :dense="!oneItem"
            :type="item.type"
            @input="closeOne(i)"
            :class="{ 'mb-0': oneItem, 'mb-1': !oneItem }"
          >
            {{ item.message }}
          </v-alert>
        </v-list-item>
      </v-slide-y-transition>
      <v-card-actions
        v-if="!oneItem"
        class="pt-0"
      >
        <v-spacer />
        <v-btn
          text
          color="primary"
          @click="close"
        >
          close
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-expand-transition>
</template>

<script>
import { mapGetters, mapState } from 'vuex'

export default {

  computed: {
    ...mapState('message', [
      'list',
    ]),
    ...mapGetters('message', [
      'top',
    ]),
    show() {
      return this.$store.getters['message/show']
    },
    oneItem() {
      return this.list.length <= 1
    },
  },

  methods: {
    close() {
      this.$store.commit('message/clear')
    },
    closeOne(i) {
      this.$store.commit('message/remove', i)
    },
  },
}
</script>

<style scoped>
.message-panel {
  position: fixed;
  z-index: 1000;
  top: 10px;

  left: 50%;
  transform: translateX(-50%);

  transition: flex 0.3s ease-out;
}
</style>
