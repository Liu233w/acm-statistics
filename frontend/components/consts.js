export const WORKER_STATUS = {
  WAITING: 'WAITING',
  WORKING: 'WORKING',
  DONE: 'DONE',
}

export const PROJECT_TITLE = 'OJ Analyzer'

export const TIMEZONE_LIST = [
  { value: 'UTC', text: '(GMT) UTC' },
  { value: 'Morocco Standard Time', text: '(GMT) Casablanca' },
  { value: 'GMT Standard Time', text: '(GMT) Greenwich Mean Time : Dublin, Edinburgh, Lisbon, London' },
  { value: 'Greenwich Standard Time', text: '(GMT) Monrovia, Reykjavik' },
  { value: 'W. Europe Standard Time', text: '(GMT+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna' },
  { value: 'Central Europe Standard Time', text: '(GMT+01:00) Belgrade, Bratislava, Budapest, Ljubljana, Prague' },
  { value: 'Romance Standard Time', text: '(GMT+01:00) Brussels, Copenhagen, Madrid, Paris' },
  { value: 'Central European Standard Time', text: '(GMT+01:00) Sarajevo, Skopje, Warsaw, Zagreb' },
  { value: 'W. Central Africa Standard Time', text: '(GMT+01:00) West Central Africa' },
  { value: 'Jordan Standard Time', text: '(GMT+02:00) Amman' },
  { value: 'GTB Standard Time', text: '(GMT+02:00) Athens, Bucharest, Istanbul' },
  { value: 'Middle East Standard Time', text: '(GMT+02:00) Beirut' },
  { value: 'Egypt Standard Time', text: '(GMT+02:00) Cairo' },
  { value: 'South Africa Standard Time', text: '(GMT+02:00) Harare, Pretoria' },
  { value: 'FLE Standard Time', text: '(GMT+02:00) Helsinki, Kyiv, Riga, Sofia, Tallinn, Vilnius' },
  { value: 'Israel Standard Time', text: '(GMT+02:00) Jerusalem' },
  { value: 'E. Europe Standard Time', text: '(GMT+02:00) Minsk' },
  { value: 'Namibia Standard Time', text: '(GMT+02:00) Windhoek' },
  { value: 'Arabic Standard Time', text: '(GMT+03:00) Baghdad' },
  { value: 'Arab Standard Time', text: '(GMT+03:00) Kuwait, Riyadh' },
  { value: 'Russian Standard Time', text: '(GMT+03:00) Moscow, St. Petersburg, Volgograd' },
  { value: 'E. Africa Standard Time', text: '(GMT+03:00) Nairobi' },
  { value: 'Georgian Standard Time', text: '(GMT+03:00) Tbilisi' },
  { value: 'Iran Standard Time', text: '(GMT+03:30) Tehran' },
  { value: 'Arabian Standard Time', text: '(GMT+04:00) Abu Dhabi, Muscat' },
  { value: 'Azerbaijan Standard Time', text: '(GMT+04:00) Baku' },
  { value: 'Mauritius Standard Time', text: '(GMT+04:00) Port Louis' },
  { value: 'Caucasus Standard Time', text: '(GMT+04:00) Yerevan' },
  { value: 'Afghanistan Standard Time', text: '(GMT+04:30) Kabul' },
  { value: 'Ekaterinburg Standard Time', text: '(GMT+05:00) Ekaterinburg' },
  { value: 'Pakistan Standard Time', text: '(GMT+05:00) Islamabad, Karachi' },
  { value: 'West Asia Standard Time', text: '(GMT+05:00) Tashkent' },
  { value: 'India Standard Time', text: '(GMT+05:30) Chennai, Kolkata, Mumbai, New Delhi' },
  { value: 'Sri Lanka Standard Time', text: '(GMT+05:30) Sri Jayawardenepura' },
  { value: 'Nepal Standard Time', text: '(GMT+05:45) Kathmandu' },
  { value: 'N. Central Asia Standard Time', text: '(GMT+06:00) Almaty, Novosibirsk' },
  { value: 'Central Asia Standard Time', text: '(GMT+06:00) Astana, Dhaka' },
  { value: 'Myanmar Standard Time', text: '(GMT+06:30) Yangon (Rangoon)' },
  { value: 'SE Asia Standard Time', text: '(GMT+07:00) Bangkok, Hanoi, Jakarta' },
  { value: 'North Asia Standard Time', text: '(GMT+07:00) Krasnoyarsk' },
  { value: 'China Standard Time', text: '(GMT+08:00) Beijing, Chongqing, Hong Kong, Urumqi' },
  { value: 'North Asia East Standard Time', text: '(GMT+08:00) Irkutsk, Ulaan Bataar' },
  { value: 'Singapore Standard Time', text: '(GMT+08:00) Kuala Lumpur, Singapore' },
  { value: 'W. Australia Standard Time', text: '(GMT+08:00) Perth' },
  { value: 'Taipei Standard Time', text: '(GMT+08:00) Taipei' },
  { value: 'Tokyo Standard Time', text: '(GMT+09:00) Osaka, Sapporo, Tokyo' },
  { value: 'Korea Standard Time', text: '(GMT+09:00) Seoul' },
  { value: 'Yakutsk Standard Time', text: '(GMT+09:00) Yakutsk' },
  { value: 'Cen. Australia Standard Time', text: '(GMT+09:30) Adelaide' },
  { value: 'AUS Central Standard Time', text: '(GMT+09:30) Darwin' },
  { value: 'E. Australia Standard Time', text: '(GMT+10:00) Brisbane' },
  { value: 'AUS Eastern Standard Time', text: '(GMT+10:00) Canberra, Melbourne, Sydney' },
  { value: 'West Pacific Standard Time', text: '(GMT+10:00) Guam, Port Moresby' },
  { value: 'Tasmania Standard Time', text: '(GMT+10:00) Hobart' },
  { value: 'Vladivostok Standard Time', text: '(GMT+10:00) Vladivostok' },
  { value: 'Central Pacific Standard Time', text: '(GMT+11:00) Magadan, Solomon Is., New Caledonia' },
  { value: 'New Zealand Standard Time', text: '(GMT+12:00) Auckland, Wellington' },
  { value: 'Fiji Standard Time', text: '(GMT+12:00) Fiji, Kamchatka, Marshall Is.' },
  { value: 'Tonga Standard Time', text: '(GMT+13:00) Nuku\'alofa' },
  { value: 'Azores Standard Time', text: '(GMT-01:00) Azores' },
  { value: 'Cape Verde Standard Time', text: '(GMT-01:00) Cape Verde Is.' },
  { value: 'Mid-Atlantic Standard Time', text: '(GMT-02:00) Mid-Atlantic' },
  { value: 'E. South America Standard Time', text: '(GMT-03:00) Brasilia' },
  { value: 'Argentina Standard Time', text: '(GMT-03:00) Buenos Aires' },
  { value: 'SA Eastern Standard Time', text: '(GMT-03:00) Georgetown' },
  { value: 'Greenland Standard Time', text: '(GMT-03:00) Greenland' },
  { value: 'Montevideo Standard Time', text: '(GMT-03:00) Montevideo' },
  { value: 'Newfoundland Standard Time', text: '(GMT-03:30) Newfoundland' },
  { value: 'Atlantic Standard Time', text: '(GMT-04:00) Atlantic Time (Canada)' },
  { value: 'SA Western Standard Time', text: '(GMT-04:00) La Paz' },
  { value: 'Central Brazilian Standard Time', text: '(GMT-04:00) Manaus' },
  { value: 'Pacific SA Standard Time', text: '(GMT-04:00) Santiago' },
  { value: 'Venezuela Standard Time', text: '(GMT-04:30) Caracas' },
  { value: 'SA Pacific Standard Time', text: '(GMT-05:00) Bogota, Lima, Quito, Rio Branco' },
  { value: 'Eastern Standard Time', text: '(GMT-05:00) Eastern Time (US & Canada)' },
  { value: 'US Eastern Standard Time', text: '(GMT-05:00) Indiana (East)' },
  { value: 'Central America Standard Time', text: '(GMT-06:00) Central America' },
  { value: 'Central Standard Time', text: '(GMT-06:00) Central Time (US & Canada)' },
  { value: 'Central Standard Time (Mexico)', text: '(GMT-06:00) Guadalajara, Mexico City, Monterrey' },
  { value: 'Canada Central Standard Time', text: '(GMT-06:00) Saskatchewan' },
  { value: 'US Mountain Standard Time', text: '(GMT-07:00) Arizona' },
  { value: 'Mountain Standard Time (Mexico)', text: '(GMT-07:00) Chihuahua, La Paz, Mazatlan' },
  { value: 'Mountain Standard Time', text: '(GMT-07:00) Mountain Time (US & Canada)' },
  { value: 'Pacific Standard Time', text: '(GMT-08:00) Pacific Time (US & Canada)' },
  { value: 'Pacific Standard Time (Mexico)', text: '(GMT-08:00) Tijuana, Baja California' },
  { value: 'Alaskan Standard Time', text: '(GMT-09:00) Alaska' },
  { value: 'Hawaiian Standard Time', text: '(GMT-10:00) Hawaii' },
  { value: 'Samoa Standard Time', text: '(GMT-11:00) Midway Island, Samoa' },
  { value: 'Dateline Standard Time', text: '(GMT-12:00) International Date Line West' },
]
