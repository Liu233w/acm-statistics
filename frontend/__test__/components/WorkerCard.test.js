import { shallowMount, createLocalVue, mount } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

jest.mock('~/dynamic/crawlers', () => () => ({
  metas: {
    testCrawler: {
      title: 'crawler title',
      description: 'crawler description',
      url: 'http://crawler.url',
    },
  },
  crawlers: {
    testCrawler: () => ({ submissions: 0, pass: 0 }),
  },
}), { virtual: true })

import { WORKER_STATUS } from '~/components/consts'
import WorkerCard from '~/components/WorkerCard.vue'
import StatisticsStore from '~/store/-dynamic/statistics'

describe('WorkerCard', () => {

  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('can be mounted correctly', () => {
    const wrapper = shallowMount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: StatisticsStore,
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('can be rendered correctly', () => {
    const wrapper = mount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: StatisticsStore,
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can render username correctly', () => {

    const wrapper = mount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: {
            state: {
              workers: [{
                crawlerName: 'testCrawler',
                username: 'test_crawler_username',
                status: WORKER_STATUS.WAITING,
                solved: 0,
                submissions: 0,
                errorMessage: '',
                tokenKey: null,
              }],
              crawlers: {
                'testCrawler': {
                  name: 'testCrawler',
                  title: 'crawler title',
                  description: 'crawler description',
                  url: 'http://crawler.url',
                },
              },
            },
            getters: StatisticsStore.getters,
            namespaced: true,
          },
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })

    expect(wrapper.vm.username).toBe('test_crawler_username')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can render warning correctly', () => {

    const wrapper = mount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: {
            state: {
              workers: [{
                crawlerName: 'testCrawler',
                username: ' username with space',
                status: WORKER_STATUS.WAITING,
                solved: 0,
                submissions: 0,
                errorMessage: '',
                tokenKey: null,
              }],
              crawlers: {
                'testCrawler': {
                  name: 'testCrawler',
                  title: 'crawler title',
                  description: 'crawler description',
                  url: 'http://crawler.url',
                },
              },
            },
            getters: StatisticsStore.getters,
            namespaced: true,
          },
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can render result correctly', () => {

    const wrapper = mount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: {
            state: {
              workers: [{
                crawlerName: 'testCrawler',
                username: 'test_crawler_username',
                status: WORKER_STATUS.WAITING,
                solved: 10,
                submissions: 20,
                errorMessage: '',
                tokenKey: null,
              }],
              crawlers: {
                'testCrawler': {
                  name: 'testCrawler',
                  title: 'crawler title',
                  description: 'crawler description',
                  url: 'http://crawler.url',
                },
              },
            },
            getters: StatisticsStore.getters,
            namespaced: true,
          },
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('can render error correctly', () => {

    const wrapper = mount(WorkerCard, {
      store: new Vuex.Store({
        modules: {
          statistics: {
            state: {
              workers: [{
                crawlerName: 'testCrawler',
                username: 'test_crawler_username',
                status: WORKER_STATUS.WAITING,
                solved: 0,
                submissions: 0,
                errorMessage: 'a error is occurred',
                tokenKey: null,
              }],
              crawlers: {
                'testCrawler': {
                  name: 'testCrawler',
                  title: 'crawler title',
                  description: 'crawler description',
                  url: 'http://crawler.url',
                },
              },
            },
            getters: StatisticsStore.getters,
            namespaced: true,
          },
        },
      }),
      localVue,
      vuetify,
      propsData: {
        index: 0,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
