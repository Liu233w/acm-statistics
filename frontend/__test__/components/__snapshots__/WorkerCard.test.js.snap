// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerCard 能正确渲染 1`] = `
<div class="v-card v-sheet theme--light">
  <header class="blue-grey lighten-5 v-sheet v-sheet--tile theme--light elevation-0 v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar" style="margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div title="crawler title" class="v-toolbar__title">
        crawler title
      </div>
      <div class="spacer"></div>
      <div class="v-toolbar__items">
        <!----> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">add_circle</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">link</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">refresh</i></span></button></span></div>
    </div>
  </header>
  <div class="container">
    <div class="layout">
      <div class="flex xs12"><span class="grey--text">
          crawler description
        </span></div>
    </div>
    <div class="layout wrap">
      <div class="flex xs12">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-36" class="v-label theme--light" style="left: 0px; position: absolute;">Username</label><input required="required" id="input-36" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--"><i role="button" class="v-icon notranslate v-icon--link material-icons theme--light"></i></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div role="dialog" class="v-dialog__container">
    <div role="document" class="v-dialog__content" style="z-index: 0;">
      <div class="v-dialog v-dialog--scrollable" style="max-width: 500px; display: none;"></div>
    </div>
  </div>
</div>
`;

exports[`WorkerCard 能正确渲染用户名 1`] = `
<div class="v-card v-sheet theme--light">
  <header class="blue-grey lighten-5 v-sheet v-sheet--tile theme--light elevation-0 v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar" style="margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div title="crawler title" class="v-toolbar__title">
        crawler title
      </div>
      <div class="spacer"></div>
      <div class="v-toolbar__items">
        <!----> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">add_circle</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">link</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">refresh</i></span></button></span></div>
    </div>
  </header>
  <div class="container">
    <div class="layout">
      <div class="flex xs12"><span class="grey--text">
          crawler description
        </span></div>
    </div>
    <div class="layout wrap">
      <div class="flex xs12">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-63" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Username</label><input required="required" id="input-63" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--clear"><i role="button" class="v-icon notranslate v-icon--link material-icons theme--light">$vuetify.icons.clear</i></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div role="dialog" class="v-dialog__container">
    <div role="document" class="v-dialog__content" style="z-index: 0;">
      <div class="v-dialog v-dialog--scrollable" style="max-width: 500px; display: none;"></div>
    </div>
  </div>
</div>
`;

exports[`WorkerCard 能正确渲染结果 1`] = `
<div class="v-card v-sheet theme--light">
  <header class="blue-grey lighten-5 v-sheet v-sheet--tile theme--light elevation-0 v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar" style="margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div title="crawler title" class="v-toolbar__title">
        crawler title
      </div>
      <div class="spacer"></div>
      <div class="v-toolbar__items">
        <!----> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">add_circle</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">link</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">refresh</i></span></button></span></div>
    </div>
  </header>
  <div class="container">
    <div class="layout">
      <div class="flex xs12"><span class="grey--text">
          crawler description
        </span></div>
    </div>
    <div class="layout wrap">
      <div class="flex xs12">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-119" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Username</label><input required="required" id="input-119" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--clear"><i role="button" class="v-icon notranslate v-icon--link material-icons theme--light">$vuetify.icons.clear</i></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div role="dialog" class="v-dialog__container">
    <div role="document" class="v-dialog__content" style="z-index: 0;">
      <div class="v-dialog v-dialog--scrollable" style="max-width: 500px; display: none;"></div>
    </div>
  </div>
</div>
`;

exports[`WorkerCard 能正确渲染警告 1`] = `
<div class="v-card v-sheet theme--light">
  <header class="blue-grey lighten-5 v-sheet v-sheet--tile theme--light elevation-0 v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar" style="margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div title="crawler title" class="v-toolbar__title">
        crawler title
      </div>
      <div class="spacer"></div>
      <div class="v-toolbar__items">
        <!----> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">add_circle</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">link</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">refresh</i></span></button></span></div>
    </div>
  </header>
  <div class="container">
    <div class="layout">
      <div class="flex xs12"><span class="grey--text">
          crawler description
        </span></div>
    </div>
    <div class="layout wrap">
      <div class="flex xs12">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-90" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Username</label><input required="required" id="input-90" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--clear"><i role="button" class="v-icon notranslate v-icon--link material-icons theme--light">$vuetify.icons.clear</i></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="layout xs12">
      <div class="flex align-self-start shrink"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light orange--text text--darken-2">warning</i></div>
      <div class="flex align-self-start"><span>用户名以空格开头</span></div>
    </div>
    <div class="layout xs12">
      <div class="flex align-self-start shrink"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light orange--text text--darken-2">warning</i></div>
      <div class="flex align-self-start"><span>用户名含有空格，部分爬虫可能不支持</span></div>
    </div>
    <!---->
  </div>
  <div role="dialog" class="v-dialog__container">
    <div role="document" class="v-dialog__content" style="z-index: 0;">
      <div class="v-dialog v-dialog--scrollable" style="max-width: 500px; display: none;"></div>
    </div>
  </div>
</div>
`;

exports[`WorkerCard 能正确渲染错误 1`] = `
<div class="v-card v-sheet theme--light">
  <header class="blue-grey lighten-5 v-sheet v-sheet--tile theme--light elevation-0 v-toolbar v-toolbar--dense v-toolbar--flat v-app-bar" style="margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
    <div class="v-toolbar__content" style="height: 48px;">
      <div title="crawler title" class="v-toolbar__title">
        crawler title
      </div>
      <div class="spacer"></div>
      <div class="v-toolbar__items">
        <!----> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">add_circle</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">link</i></span></button></span> <span class="v-tooltip v-tooltip--bottom"><div class="v-tooltip__content" style="left: 0px; max-width: auto; opacity: 0; top: 12px; z-index: 0; display: none;"></div><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light">refresh</i></span></button></span></div>
    </div>
  </header>
  <div class="container">
    <div class="layout">
      <div class="flex xs12"><span class="grey--text">
          crawler description
        </span></div>
    </div>
    <div class="layout wrap">
      <div class="flex xs12">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-146" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Username</label><input required="required" id="input-146" type="text"></div>
              <div class="v-input__append-inner">
                <div class="v-input__icon v-input__icon--clear"><i role="button" class="v-icon notranslate v-icon--link material-icons theme--light">$vuetify.icons.clear</i></div>
              </div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div role="dialog" class="v-dialog__container">
    <div role="document" class="v-dialog__content" style="z-index: 0;">
      <div class="v-dialog v-dialog--scrollable" style="max-width: 500px; display: none;"></div>
    </div>
  </div>
</div>
`;
